aspnetcore/signalr/publish-to-azure-web-app.md

```diff
@@ -1,63 +1,80 @@
+1. Right-click on the project in **Solution Explorer** and select **Publish**. Confirm that **App Service** and **Create new** are selected in the **Pick a publish target** dialog. Select **Create Profile** from the **Publish** button drop down.
 
-![Pick publish target](publish-to-azure-web-app/_static/pick-publish-target-dialog.png)
+   Enter the information described in the following table in the **Create App Service** dialog and select **Create**.
 
-Enter the following information in the **Create App Service** dialog and select **Create**.
+   | Item               | Description |
+   | ------------------ | ----------- |
+   | **Name**           | Unique name of the app. |
+   | **Subscription**   | Azure subscription that the app uses. |
+   | **Resource Group** | Group of related resources to which the app belongs. |
+   | **Hosting Plan**   | Pricing plan for the web app. |
 
-| Item | Description |
-| ---- | ----------- |
-| **App name** | A unique name of the app. |
-| **Subscription** | The Azure subscription that the app uses. |
-| **Resource Group** | The group of related resources to which the app belongs.  |
-| **Hosting Plan** | The pricing plan for the web app. |
+1. Select the **Azure SignalR Service** in the **Dependencies** > **Add** drop-down list:
 
-![Create app service](publish-to-azure-web-app/_static/create-app-service-dialog.png)
+   ![Dependencies area showing the selection of Azure SignalR Service in the Add drop-down list](publish-to-azure-web-app/_static/signalr-service-dependency.png)
+
+1. In the the **Azure SignalR Service** dialog, select **Create a new Azure SignalR Service instance**. Provide a **Name**, **Resource Group**, and **Location**. Return to the **Azure SignalR Service** dialog and select **Add**.
 
 Visual Studio completes the following tasks:
 
 * Creates a Publish Profile containing publish settings.
 * Creates or uses an existing *Azure Web App* with the provided details.
 * Publishes the app.
-* Launches a browser, with the published web app loaded.
+* Launches a browser, which loads the web app.
+
+The format of the app's URL is `{App Service Name}.azurewebsites.net`. For example, an app named `SignalRChatApp` has a URL of `https://signalrchatapp.azurewebsites.net`.
+
+If an HTTP *502.2 - Bad Gateway* error occurs during a preview release, see [Deploy ASP.NET Core preview release to Azure App Service](xref:host-and-deploy/azure-apps/index#deploy-aspnet-core-preview-release-to-azure-app-service) to resolve it.
 
-Notice the format of the URL for the app is *{app name}.azurewebsites.net*. For example, an app named `SignalRChattR` has a URL that looks like `https://signalrchattr.azurewebsites.net`.
+## Configure the app in Azure App Service
 
-If an HTTP 502.2 error occurs, see [Deploy ASP.NET Core preview release to Azure App Service](xref:host-and-deploy/azure-apps/index) to resolve it.
+*This section only applies to apps not using the Azure SignalR Service.*
```

Comment:
But the directions above say to specify Azure SignalR Service as a dependency.

Reply:
Yes, but they aren't required to do so. If they don't, then they need to follow the guidance in this section. Do you have alternative wording?

Reply:
My suggestion is to make the earlier step explicitly optional.  Would be ideal to give guidance on why they might choose not to do it, but that can wait for another day.

Reply:
@bradygaster This is really your :baby: ... go ahead with making the service explicitly optional? Scares me a little bit. 🎃

I'm interested in the following cases:

* Existing Azure App Service
* Existing Publish Profile

Reply:
If Az SignalR Service is not optional, we should be able to remove the "what to do if you didn't select Az SignalR Service" section later in the article.